### ISSUE TITLE
crazy installation idea

### ISSUE THREAD

**Issue by kylestetz** [punkave/zoltar#12]
I'm thinking about the most helpful installation process, given the fact that you need to open your network prefs and paste in a file path. What if:

when I `npm install -g zoltar` I get a message that says

```
Zoltar is almost ready. Run `zoltar setup` to finish installation.
```

Here's what `zoltar setup` could do in OS X (no idea what this might look like elsewhere):

`pbcopy` the filepath of that `.pac` file to your clipboard, output instructions on where to paste it, and then run `open /System/Library/PreferencePanes/Network.prefPane/` which opens the network preferences, getting you about halfway towards pasting the filepath.

There's something nice about keeping the instructions in the command line during the process, rather than having to reference the github readme.

**Comment by boutell** [comment]
True, and perhaps there's a command line way to finish that job, who
knows...

On Thu, Dec 12, 2013 at 2:46 PM, Kyle Stetz notifications@github.comwrote:

> I'm thinking about the most helpful installation process, given the fact
> that you need to open your network prefs and paste in a file path. What if:
> 
> when I npm install -g zoltar I get a message that says
> 
> Zoltar is almost ready. Run `zoltar setup` to finish installation.
> 
> Here's what zoltar setup could do in OS X (no idea what this might look
> like elsewhere):
> 
> pbcopy the filepath of that .pac file to your clipboard, output
> instructions on where to paste it, and then run open
> /System/Library/PreferencePanes/Network.prefPane/ which opens the network
> preferences, getting you about halfway towards pasting the filepath.
> 
> There's something nice about keeping the instructions in the command line
> during the process, rather than having to reference the github readme.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/punkave/zoltar/issues/12
> .

## 

Tom Boutell
P'unk Avenue
215 755 1330
punkave.com
window.punkave.com

**Comment by etodanik** [comment]
This is the file you'll need to look at:
/Library/Preferences/SystemConfiguration/preferences.plist

**Comment by etodanik** [comment]
Or better yet, the magic CLI line in OSX you're looking for is:

```
networksetup -setautoproxyurl Wi-Fi "file://localhost/Users/username/src/zoltar/proxy.pac" 
```

(You can replace Wi-Fi with Ethernet too)

You can use the following to be smart, and know the current state of that configuration value (to prevent overwrite in case the user has a setting in place):

```
networksetup -getautoproxyurl Wi-Fi
```

Ideally you wouldn't even do it at setup time, but at run time, instead of the warning about proxy.pac

**Comment by etodanik** [comment]
Hey guys,

I added a quick pull request for this feature under OSX. 

https://github.com/punkave/zoltar/pull/14
